components/director/internal/repo/conditions.go

```diff
@@ -0,0 +1,114 @@
+package repo
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Condition interface {
+	// GetQueryPart returns formatted string that will be included in the SQL query for a given condition
+	GetQueryPart() string
```

Comment:
I think about returns here also the `args` as `interfacae{}` and have only one method `GetCondition`

Reply:
Hmm, but anyway `getAllArgs` and `writeEnumeratedConditions` use only query or arguments. I would need to refactor these functions as well, so maybe I will stop here with further refactoring :)